### PR TITLE
Fixes DOCS-176 by adding info on batch ingestion job types

### PR DESCRIPTION
**Pull request by matthewhelmke**
Fixes [DOCS-176](https://cortexdata.atlassian.net/browse/DOCS-176).
This is a simple addition of information provided to a customer via Slack, just a copy/paste/edit.

### CODE REVIEW

File: configuration-reference/job-specification.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ The following configurations are supported by Pinot
...
...e>SegmentCreation</code></li><li><code>SegmentTarPush</code></li><li><code>SegmentUriPush</code></li><li><code>SegmentMetadataPush</code></li><li><code>SegmentCreationAndTarPush</code>: When you are using NFS or similar that sits behind a controller and you are unable to externally copy segments there, this pushes the segment payload.</li><li><code>SegmentCreationAndUriPush</code>: When using deepstore, this creates the segment on the deepstore and pushes the URI to the controller, the controller then downloads the segment to extract the metadata from the URI and copies to the final destination.</li><li><code>SegmentCreationAndMetadataPush</code>: When using deepstore, this is the most efficient as the segment payload does not have to pass through the controller.</li></ul><p><strong>Note:</strong> For production environments where Pinot Deep Store is configured, it's recommended to use <strong>SegmentCreationAndMetadataPush</strong></p>                                                                                                                                                                                                                                                         |
```
Comment (kelseiv) [inline configuration-reference/job-specification.md]:
```suggestion
| jobType   | <p>Type of job to execute. The following types are supported</p><ul><li><code>SegmentCreation</code></li><li><code>SegmentTarPush</code></li><li><code>SegmentUriPush</code></li><li><code>SegmentMetadataPush</code></li><li><code>SegmentCreationAndMetadataPush</code>: (Recommended for production environments where Pinot deep store is configured): Use this job to bypass the controller, and send the segment payload directly to the data store.</li><li><code>SegmentCreationAndUriPush</code>: (Alternative option if Pinot deep store is configured) Use this job to create the segment on the deep store, push the URI to the controller to download the segment, extract metadata from the URI, and copy the data to deep store.</li><li><code>SegmentCreationAndTarPush</code>: If you use Network File System (NFS) or something that sits behind a controller, and are unable to externally copy segments to the data store, use this job to push the segment payload.</li>
```

Comment (kelseiv) [reply configuration-reference/job-specification.md]: unclear how the first 4 jobs (SegmentCreation, SegmentTarPush, SegmentUriPush, SegmentMetadataPush) align with the last 3 jobs with descriptions (SegmentCreationAndMetadataPush, SegmentCreationAndUriPush, SegmentCreationAndTarPush)

if we are describing the same jobs here (and maybe there's a typo?), think we can combine into one list

**Review by kelseiv — COMMENTED** [review]
- Question about whether there are 3 or 4 jobs here, perhaps for Mayank?
- Reworded text a bit
- Resorted options to make the recommended option first, and then the other option that requires deep store second. Also curious, for the second deep store option, what cases would they use this rather than the recommended more efficient option?

File: configuration-reference/job-specification.md (hunk 1 above)
Comment (matthewhelmke) [reply configuration-reference/job-specification.md]: There are 7 jobs total. Four of them involve a single action. The last three are combinations of the first and one of #s 2, 3, or 4. So, the list is solid. Looking through your other suggestions expecting they will be valuable. :)

Comment (kelseiv) [reply configuration-reference/job-specification.md]: ah, I see, perhaps it'd be more clear/parallel if they all had descriptions?

File: configuration-reference/job-specification.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ The following configurations are supported by Pinot
...
...e>SegmentCreation</code></li><li><code>SegmentMetadataPush</code></li><li><code>SegmentTarPush</code></li><li><code>SegmentUriPush</code></li><li><code>SegmentCreationAndMetadataPush</code>: When using deepstore, this is the most efficient as the segment payload does not have to pass through the controller.</li><li><code>SegmentCreationAndTarPush</code>: When you are using NFS or similar that sits behind a controller and you are unable to externally copy segments there, this pushes the segment payload.</li><li><code>SegmentCreationAndUriPush</code>: When using deepstore, this creates the segment on the deepstore and pushes the URI to the controller, the controller then downloads the segment to extract the metadata from the URI and copies to the final destination.</li></ul><p><strong>Note:</strong> For production environments where Pinot Deep Store is configured, it's recommended to use <strong>SegmentCreationAndMetadataPush</strong></p>                                                                                                                                                                                                                                                         |
```
Comment (kelseiv) [inline configuration-reference/job-specification.md]: I think we should tweak the passive note, and clean up the grammar--I added suggestions [here](https://github.com/pinot-contrib/pinot-docs/pull/252/files/1dd1cd0659a87770bff8c091480366d7f71b63d6#r1366218111)

Comment (matthewhelmke) [reply configuration-reference/job-specification.md]: Edited, thank you!

**Pull request merged**